src/node/transactions.rs

```diff
@@ -46,14 +46,28 @@ impl NodeClient {
             Ok(Response::Rejected(reason)) => {
                 let haskell_display = as_node_submit_error(reason);
                 info!(
-                    "N2C[{}]: {}: {:?}",
-                    self.connection_id, "TxSubmitFail", haskell_display
+                    "N2C[{}]: {}: {}, CBOR: {}",
+                    self.connection_id,
+                    "TxSubmitFail",
+                    haskell_display,
+                    hex::encode(&tx)
                 );
                 Err(BlockfrostError::custom_400(haskell_display))
             },
             Err(e) => {
-                let error_message = format!("Error during transaction submission: {:?}", e);
+                let error_message = format!(
+                    "Error during transaction submission: {:?}, CBOR: {}",
+                    e,
+                    hex::encode(&tx)
+                );
                 self.invalidate_connection(&error_message); // Never use this connection again.
+                error!(
```

Comment:
I _think_ (haven’t checked) that you'll get 3 error messages for the same event:
1. From `self.invalidate_connection`
2. From this `error!`
3. From the error middleware for the `BlockfrostError`

Maybe it’s fine for now

Reply:
Yes, we are already getting twice of each error :(.
We will need to tidy up the logs in the future.
But for now, over-logging is better than under-logging in my opinion.